### ISSUE TITLE
Columns broken when testing output via MailChimp

### ISSUE THREAD

**Issue by danerich** [mjmlio/mjml#844]
Hey, 

We have been experiencing a few issues relating to #717 with columns displaying at 100% width on Gmail webmail even tho media queries are supported in the offending email client.

I was experiencing the issue in litmus builder until I realized the inline CSS option was on. Switching that of fixed that issue.

When sending via MailChimp using the uploaded zip option (useful for uploading images for MC to host and relink) 

Test MJML: https://gist.github.com/danerich/37411268e8e8111bad70aa636b02287f
HTML from MJML: https://gist.github.com/danerich/73e78c8a552bc058cf834c1e35a0bfdb
MailChimp Output: https://gist.github.com/danerich/d30efd2a6ef7886b0917ea1e77db9479

I can confirm that MailChimp is altering the code It looks to mess with the styling and remove do a wired repetition of the media queries. It also removes:
```
<!--[if !mso]><!-->
<style type="text/css">
  @media only screen and (max-width:480px) {
    @-ms-viewport { width:320px; }
    @viewport { width:320px; }
  }
</style>
<!--<![endif]-->
```
Removeing the repeated CSS and adding in the above looks to fix the issue in gmail and the columns work.

Confirmation from MailChimp on the altering the code: 
> Great - got it. I confirmed with MailChimp support that the web app does use a filter function that moves all CSS to the head, then on send, inlines it. The filter also nukes conditional comments, most likely because they look like MailChimp specific tokens.

https://groups.google.com/forum/#!topic/mailchimp-api-discuss/SlJu13KdXsM

I understand that there is a workaround which involves using the MailChimp paste in code option instead as it looks like MailChimp does not alter the code when using this method.

I am just concerned if we were to supply code and the images in a ZIP to the client for them to send via there ESP it would break also. It would be great if we could find a way to stop MailChimp and other inliners breaking the email.

**Comment by iRyusa** [comment]
Hi @danerich  

Well I understand the concern, but here it's clearly Mailchimp fault to totally mess with the HTML.
I don't really remember what does this MQ  but I think it's something about Windows Phone & IE ?
So, I'm not sure what could be done in our side.

You might need to ping them to get someone working on this issue

**Comment by danerich** [comment]
I agree this is their problem.

I did contact them and got the following response:

> Thank you for providing me with a link to the thorough explanation. Just to confirm, are you asking if there is anyway to prevent Mailchimp's code purification process? It will remove certain styling or other aspects of code that may interfere negatively with formatting.  
> 
> There is not a way for us to disable the way code is purified when importing files. However, if you feel this may be a limitation when sending the files to your client, I would be happy to pass this onward as feedback. 

I went back to them saying it would be great if we could switch this off so the code is left intact as indented. Having to second guess how their undocumented ‘code purification process’ will change the HTML adds an extra layer of complication to an already complicated email development landscape.

The support person has set this on as feedback. I asked for further updates if there are any. I will update if I hear anything else from them.

**Comment by dalefish** [comment]
They gave me the same answer a year ago, don't hold your breath :)

Kind Regards,

:----------------------------:
Dale McConnell

**Comment by dermotbrennan** [comment]
In my case, I discovered that the MailChimp design editor was prettifying my minified HTML and adding in whitespace. 

I had been pasting in my minified HTML into the design editor, saving and sending a test and it was broken in the Gmail iOS app.

I eventually updated the content of the campaign using the API and it then sent the original HTML unmodified. (I also ensured that the CSS inliner option was unchecked).

My advice would be to not use the MailChimp design editor if you have very optimised HTML.
